File: socialuni-web/src/main/java/com/socialuni/center/web/model/VO/FrontErrorLogVO.java
package com.socialuni.center.web.model.VO;

import com.socialuni.center.web.model.DO.config.FrontErrorLogDO;
import com.socialuni.center.web.utils.DevAccountUtils;
import com.socialuni.entity.model.DO.user.UserDO;
import lombok.Data;

import java.util.Date;

/**
 * @author qinkaiyuan
 * @date 2018-11-18 20:48
 */
//记录前端错误日志，不该出现的错误
@Data
public class FrontErrorLogVO {
    private UserDO user;

    //出错的url
    private String uri;

    //出错的业务备注
    private String detail;

    //传入了什么参数导致报错
    private String params;

    //错误信息
    private String errorMsg;
    private String platform;
    private String provider;
    private String appVersion;

    public FrontErrorLogDO toDO(UserDO user) {
        FrontErrorLogDO front = new FrontErrorLogDO();
        front.setCreateTime(new Date());
        if (user != null) {
            front.setUserId(user.getId());
        }
        front.setDetail(detail);
        front.setUri(uri);
        front.setParams(params);
        front.setErrorMsg(errorMsg);
        front.setPlatform(platform);
        front.setProvider(provider);
        front.setAppVersion(appVersion);
        front.setDevId(DevAccountUtils.getDevId());
        return front;
    }
}

File: socialuni-js/packages/socialuni-api/socialuni-api-base/src/model/FrontErrorLogVO.ts
import PlatformType from "socialuni-constant/constant/PlatformType";
import {socialuniSystemModule} from "qing-util/src/store/SocialuniSystemModule";

export default class FrontErrorLogVO {
    uri: string = null
    detail: string = null
    params: string = null
    errorMsg: string = null
    platform: string = null
    provider: string = null
    frontPage: string = null
    appVersion: number = null

    constructor(uri: string, detail: string, params: string, errorMsg: string, frontPage: string) {
        this.uri = uri
        this.detail = detail
        this.params = params
        this.frontPage = frontPage
        this.errorMsg = errorMsg
        this.platform = socialuniSystemModule.platform
        if (this.platform === PlatformType.mp) {
            this.provider = socialuniSystemModule.mpPlatform
        } else {
            this.appVersion = socialuniSystemModule.appVersion
        }
    }
}
